core/Validators/CharacterLength.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Validators;
+
+use Piwik\Common;
+use Piwik\Piwik;
+
+class CharacterLength extends BaseValidator
+{
+    /**
+     * @var null|int
+     */
+    private $min;
+
+    /**
+     * @var null|int
+     */
+    private $max;
+
+    /**
+     * @param null|int $min
+     * @param null|int $max
+     */
+    public function __construct($min = null, $max = null)
+    {
+        if (isset($min)) {
+            $this->min = (int) $min;
+        }
+        if (isset($max)) {
+            $this->max = (int) $max;
+        }
+    }
+
+    public function validate($value)
+    {
+        if (is_array($value) || is_object($value) || is_resource($value)) {
+            return;
+        }
```

Comment:
Might be better here indeed in case a value is maybe false or null I reckon. I remember there was a reason I didn't use it but can't think of it now and don't think it makes too much sense. So will change it 👍